packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts

```diff
@@ -196,10 +204,50 @@ export default class CallDiagnosticMetrics {
   }
 
   /**
-   * TODO: NOT IMPLEMENTED
+   * Return Client Event payload by client error code
+   * @param clientErrorCode
+   * @returns
+   */
+  public getErrorPayloadForClientErrorCode(clientErrorCode: number): ClientEventError {
+    let error: ClientEventError;
+
+    if (clientErrorCode) {
+      const partialParsedError = CLIENT_ERROR_CODE_TO_ERROR_PAYLOAD[clientErrorCode];
+
+      if (partialParsedError) {
+        error = merge(
+          {fatal: true, shownToUser: false, name: 'other', category: 'other'}, // default values
+          partialParsedError
+        );
+
+        return error;
+      }
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Generate error payload for Client Event
    * @param rawError
    */
-  generateErrorPayload(error: any) {
+  generateClientEventErrorPayload({rawError}: {rawError?: any}) {
```

Comment:
for a single parameter you do not need an object param
```suggestion
  generateClientEventErrorPayload(rawError?: any) {
```
but if you keep it then please update the JSDocs

Reply:
refactored